storage/innobase/srv/srv0start.cc

```diff
@@ -2689,6 +2691,20 @@ dberr_t srv_start(bool create_new_db, const std::string &scan_directories) {
 
       log_sys_close();
 
+      // We have set mysql's dd flags based on what we read from page 0. This
+      // was before redo log was applied. Now we have applied changes from redo
+      // log. If there were changes in redo for page 0 of mysql.ibd we check if
+      // encryption flag has changed. If it did we amend mysql's dd encryption
+      // flags accordingly.
+      if (mysql_ibd_page_0_present_in_redo) {
+        fil_space_t *mysql_space = fil_space_get(dict_sys_t::s_space_id);
+        if (mysql_space == nullptr ||
+            dd_fix_mysql_ibd_encryption_flag_if_needed(current_thd,
```

Comment:
you are not fixing the fil_space_t flags of mysql tablespace. Only fixing DD is not sufficient.

Reply:
mysql tablespace fil_space_t flags was already updated by redo log applier.

Reply:
mysql tablespace fil_space_t flags was already updated by redo log applier.

Reply:
Ok, is it possible to add assertion that flags have encrypted bit set?

Reply:
Why encrypted bit set ? it might be that the bit is not set. It's checking if mysql table was changed by redo applier and then it sets encryption flag in DD to the one from fil_space_t.

Reply:
Ok, I see